runtime/vm/createramclass.cpp

```diff
@@ -3108,6 +3108,15 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
 			if (J9UTF8_LITERAL_EQUALS(J9UTF8_DATA(className), J9UTF8_LENGTH(className), MAGIC_ACCESSOR_IMPL)) {
 				classFlags |= J9ClassIsExemptFromValidation;
 			}
+
+#if JAVA_SPEC_VERSION == 8
+			/* check if a proxy class is being created by checking if the name matches the generated implementation class
+			   name excluding the generated characters */
+			if (0 == strncmp( (char*)J9UTF8_DATA(className), "com/sun/proxy/$Proxy", 20)) {
```

Comment:
nits: no space after the opening parenthesis; the cast should be `(const char *)`.

What prevents a hostile class from defining classes with this name prefix and bypassing validation?